apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -1,47 +1,184 @@
+      lastMessageType: thread.lastMessageType,
+      draft: Drafts.byThreadId(thread.id)
+    };
+  }
 
   var ConversationService = {
     /**
+     * List of mobileMessageClients mapped to specific application id.
+     * @type {Map.<string, Client>}
+     */
+    [priv.mobileMessageClients]: null,
+
+    /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this[priv.mobileMessageClients] = new Map();
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
+     * @param {appInstanceId} appInstanceId Unique id of the app instance
+     * requesting service.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream, appInstanceId) {
+      mark('start retrieving conversations');
+
+      var getThreadsStream = this[priv.getMobileMessageClient](
+        appInstanceId
+      ).stream('getThreads');
+
+      // If conversation stream is cancelled we should cancel internal threads
+      // stream as well.
+      serviceStream.cancel = (reason) => {
+        debug('getAllConversations stream is cancelled: %s', reason);
+        return getThreadsStream.cancel(reason);
+      };
+
+      var draftsPromise = Drafts.request().then(() => {
+        // Return draft list that is sorted by timestamp in inverse order.
+        return Drafts.getAllThreadless().sort(
+          (draftA, draftB) => draftB.timestamp - draftA.timestamp
+        );
+      });
+
+      // Assume that we get the data in an inverse sort order.
+      var index = 0;
+      getThreadsStream.listen((thread) => {
+        mark(++index + ' conversation retrieved');
+        debug('Retrieved conversation with id %s', thread.id);
+
+        draftsPromise.then((drafts) => {
+          // Flush drafts that are created earlier than current thread.
+          while(drafts.length && drafts[0].timestamp >= thread.timestamp) {
+            serviceStream.write(draftToConversationSummary(drafts.shift()));
+          }
+
+          serviceStream.write(threadToConversationSummary(thread));
+        });
+      });
+
+      return getThreadsStream.closed.then(() => draftsPromise).then(
+      (drafts) => {
+        for(var draft of drafts) {
+          serviceStream.write(draftToConversationSummary(draft));
+        }
+
+        mark('All conversations retrieved');
+
+        serviceStream.close();
+      }).catch((e) => {
+        // Pass error object once the following issue is fixed:
+        // https://github.com/gaia-components/threads/issues/74
+        serviceStream.abort(e.message || e.name);
```

Comment:
Why not :) What about template string ```[${e.name}] ${e.message || ''}```, it looks simpler for the cost of additional space in case we don't have ```message```. Does it work for you?